pkgs/applications/networking/browsers/chromium/browser.nix

```diff
@@ -38,6 +38,25 @@ mkChromiumDerivation (base: rec {
       mkdir -vp "$logo_output_path"
       cp -v "$icon_file" "$logo_output_path/$packageName.png"
     done
+
+    # Install Desktop Entry
+    install -D chrome/installer/linux/common/desktop.template \
+      $out/share/applications/chromium-browser.desktop
+
+    substituteInPlace $out/share/applications/chromium-browser.desktop \
+      --replace "@@MENUNAME@@" "Chromium" \
+      --replace "@@PACKAGE@@" "chromium" \
+      --replace "Exec=/usr/bin/@@USR_BIN_SYMLINK_NAME@@" "Exec=chromium"
+
+    # Append more mime types to the end
+    sed -i '/^MimeType=/ s/$/x-scheme-handler\/webcal;x-scheme-handler\/mailto;x-scheme-handler\/about;x-scheme-handler\/unknown/' \
+      $out/share/applications/chromium-browser.desktop
+
+    # See https://github.com/NixOS/nixpkgs/issues/12433
+    sed -i \
+      -e '/\[Desktop Entry\]/a\' \
+      -e 'StartupWMClass=chromium-browser' \
```

Comment:
Would renaming the desktop file to `chromium.desktop` allow us to get rid of the `StartupWMClass`. If I recall correctly, though I cannot find the reference other than [this](https://stackoverflow.com/a/11851378/160386) now, it should get the default value from the program name.

Though weirdly, `xprop WM_CLASS` says the window is indeed `chromium-browser`. Are we setting it somewhere like Gentoo does according to [this](https://superuser.com/questions/488391/chromium-always-starts-as-floating-in-awesome-wm#comment580617_488393)?

Reply:
I think you might be right, Gentoo actually does the same thing as we:
https://gitweb.gentoo.org/repo/gentoo.git/tree/www-client/chromium/chromium-69.0.3497.92.ebuild?id=cccb1ca7f6e9d63f53a9ed3bbe1b34c3c9945a6d#n653

Reply:
No, if I did everything right, that doesn't help.
I can't quite figure out what we should do from those conversations:
https://bugs.chromium.org/p/chromium/issues/detail?id=174794
https://bugzilla.gnome.org/show_bug.cgi?id=673657

Seems like what we're doing is wrong. But it seems to be working for us (and Gentoo). I don't know how to fix it because chromium never seems to have `chromium` as it's `WM_CLASS`.

So I propose that we just keep things like they are because they seem to be working fine.

Reply:
Hmm, it looks like the long name is hard-coded in the file: `strings $(nix-build -A chromium.browser)/libexec/chromium/chromium | grep chromium-browser`

I am not sure where does it come from. But [Arch package](https://www.archlinux.org/packages/extra/x86_64/chromium/) has the same thing, even though they do not use `StartupWMClass`.

Yeah, let’s keep things like they are for now.

Reply:
I think it might come from here: https://cs.chromium.org/chromium/src/chrome/installer/linux/common/chromium-browser/chromium-browser.info